modules/packages/apk

```diff
@@ -0,0 +1,91 @@
+#!/bin/sh -e
+get_package_data() {
+  name="${INPUT_File?File must be given to get-package-data}"
+  echo PackageType=repo
+  echo Name=$name
+}
+
+list_installed() {
+  # Example pkg output:
+  # Name         Version  Rev   Developer  Notes
+  # core         16-2.30  3748  canonical  core
+  # hello-world  6.3      27    canonical  -
+  #
+  # After rewrite:
+  # Name=core
+  # Version=16-2.30
+  # Architecture=none
+  apk list --installed | sed 's/-/ /' | awk '
+{
+    printf("Name=%s\n",$1)
+    printf("Version=%s\n",$2)
+    printf("Architecture=%s\n",$3)
+}'
+}
+
+repo_install() {
```

Comment:
I see repo_installl, what about file_install if I have an apk locally on the file system?

Reply:
have a look at `get_package_data()` function - it unconditionally considers all packages to be coming from repo - so installation from files is not supported.

Reply:
I commented elsewhere about this. get_package_data() would need to be refined. ack.

Reply:
implemented below. :+1: